tests/server/dataSource-tests.js

```diff
@@ -0,0 +1,606 @@
+                    "rev",
+                    "{arguments}.0._rev"
+                ]
+            }, {
+                task: "{deleteStoryDataSource}.set",
+                args: [{
+                    directStoryId: "story1",
+                    directRevisionId: "invalid_rev"
+                }],
+                reject: "sjrk.test.storyTelling.server.dataSource.tester.assertMissingError",
+                rejectArgs: [
+                    "Error thrown for invalid story rev",
+                    "{arguments}.0",
+                    {
+                        storyId: "story1",
+                        rev: "invalid_rev",
+                        method: "DELETE"
+                    }
+                ]
+            }, {
+                task: "{deleteStoryDataSource}.set",
+                args: [{
+                    directStoryId: "invalid_story_id",
+                    directRevisionId: "{that}.rev"
+                }],
+                reject: "sjrk.test.storyTelling.server.dataSource.tester.assertMissingError",
+                rejectArgs: [
+                    "Error thrown for invalid story id",
+                    "{arguments}.0",
+                    {
+                        storyId: "invalid_story_id",
+                        rev: "{that}.rev",
+                        method: "DELETE"
+                    }
+                ]
+            }, {
+                task: "{deleteStoryDataSource}.set",
+                args: [{
+                    directStoryId: "story1",
+                    directRevisionId: "{that}.rev"
+                }],
+                resolve: "sjrk.test.storyTelling.server.dataSource.tester.assertResponse",
+                resolveArgs: [
+                    "Story should have been removed",
+                    {
+                        "ok": true,
+                        "id": "story1"
+                    },
+                    "{arguments}.0"
+                ]
+            }, {
+                task: "{storyDataSource}.get",
+                args: [{directStoryId: "story1"}],
+                reject: "sjrk.test.storyTelling.server.dataSource.tester.assertMissingError",
+                rejectArgs: [
+                    "Error thrown for attempting to access a deleted story",
+                    "{arguments}.0",
+                    {storyId: "story1"}
+                ]
+            }]
+        }]
+    }]
+});
+
+/**
+ * A wrapper around `jqUnit.assertDeepEq` except it strips `_rev` out of the actual response as this is generated value.
+ * The remaining properties are compared with `jqUnit.assertDeepEq`.
+ *
+ * @param {String} msg - The assertion message
+ * @param {Object} expected - The expected value to compare against the `actual` value.
+ * @param {Object} actual - The actual value, minus `_rev` property, compared against the `expected` value.
+ */
+sjrk.test.storyTelling.server.dataSource.tester.assertResponse = function (msg, expected, actual) {
```

Comment:
A similar filtering out of `_rev` is done in `sjrk.storyTelling.server.testServerWithStorageDefs.verifyStoryPersistence`. Perhaps that piece of code and this function could be combined in a shared function in `serverTestUtils.js`.